samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -318,8 +327,26 @@ def on_after_transform_template(self, template):
                             TemplateId=self._sanitize_sar_str_param(template_id),
                         )
                     )
-                    response = self._sar_service_call(get_cfn_template, application_id, application_id, template_id)
+                    response = {"Status": "PREPARING"}  # default response if we can't reach SAR
+
+                    try:
+                        if not throttled:
+                            response = self._sar_service_call(
+                                get_cfn_template, application_id, application_id, template_id
+                            )
+                    except ClientError as e:
+                        error_code = e.response["Error"]["Code"]
+                        if error_code == "TooManyRequestsException":
```

Comment:
Why only catch this here? Can't this happen in the `create_cloud_formation_template`?

Reply:
It probably can -- we just haven't seen it in any availability dips.  There's no loop around the create at the moment so we'd either have to add a loop (to `_sar_service_call` if we want it to be common) or consider throttling an error.